src/basic/parse-util.h

```diff
@@ -136,6 +135,14 @@ int parse_ip_prefix_length(const char *s, int *ret);
 
 int parse_oom_score_adjust(const char *s, int *ret);
 
+/* Implement floating point using fixed integers, to improve performance when
+ * calculating load averages. These macros can be used to extract the integer
+ * and decimal parts of a value. */
+#define PRECISION_BITS  11
+#define FIXED_POINT_1_0 (1 << PRECISION_BITS)
+#define INT_SIDE(x)     ((x) >> PRECISION_BITS)
+#define DECIMAL_SIDE(x) INT_SIDE(((x) & (FIXED_POINT_1_0 - 1)) * 100)
```

Comment:
if we now use our own definitions of this, maybe turn these into typesafe inline functions?

Also, please prefix with `LOADAVG_`/`loadvg_` or so, given that the 11bit precision is a loadvg thing, nothing else.

Reply:
> if we now use our own definitions of this, maybe turn these into typesafe inline functions?

I wanted to keep the implementation 1:1 to minimize risk, since I'm not that familiar with oomd and I just wanted to solve the license issue

Reply:
for the prefixes: https://github.com/systemd/systemd/pull/20854